src/main/java/cloud/fogbow/ras/core/plugins/interoperability/aws/quota/v2/AwsV2ComputeQuotaPlugin.java

```diff
@@ -1,40 +1,244 @@
-                Integer.valueOf(properties.getProperty(AwsConstants.RAM_QUOTA)),
-                Integer.valueOf(properties.getProperty(AwsConstants.INSTANCES_QUOTA))
-        );
-
-        return new ComputeQuota(totalQuota, usedQuota);
-    }
+	private static final Logger LOGGER = Logger.getLogger(AwsV2ComputeQuotaPlugin.class);
+
+	private static final String DEFAULT_INSTANCE_TYPE = "t1.micro";
+	private static final String COMMENTED_LINE_PREFIX = "#";
+	private static final String CSV_COLUMN_SEPARATOR = ",";
+
+	private static final int INSTANCE_TYPE_COLUMN = 0;
+	private static final int LIMITS_COLUMN = 11;
+	private static final int MAXIMUM_STORAGE_VALUE = 300;
+	private static final int MEMORY_COLUMN = 2;
+	private static final int ONE_GIGABYTE = 1024;
+	private static final int ONE_TERABYTE = 1000;
+	private static final int VCPU_COLUMN = 1;
+
+	private Map<String, ComputeAllocation> allocatedInstancesMap;
+	private Map<String, ComputeAllocation> availableInstanceMap;
+	private String flavorsFilePath;
+	private String region;
+
+	public AwsV2ComputeQuotaPlugin(String confFilePath) {
+		Properties properties = PropertiesUtil.readProperties(confFilePath);
+		this.region = properties.getProperty(AwsV2ConfigurationPropertyKeys.AWS_REGION_SELECTION_KEY);
+		this.flavorsFilePath = properties.getProperty(AwsV2ConfigurationPropertyKeys.AWS_FLAVORS_TYPES_FILE_PATH_KEY);
+		this.allocatedInstancesMap = new HashMap<String, ComputeAllocation>();
+		this.availableInstanceMap = new HashMap<String, ComputeAllocation>();
+	}
+
+	@Override
+	public ComputeQuota getUserQuota(AwsV2User cloudUser) throws FogbowException {
+		updateAllocationsMap();
+		Ec2Client client = AwsV2ClientUtil.createEc2Client(cloudUser.getToken(), this.region);
+		Map<String, ComputeQuota> quotas = loadQuotasByInstances(client);
+		return calculateQuotas(quotas);
+	}
+
+	private ComputeQuota calculateQuotas(Map<String, ComputeQuota> quotas) {
+		ComputeAllocation totalQuota;
+		ComputeAllocation usedQuota;
+		ComputeQuota result = null;
+		int totalDisk = MAXIMUM_STORAGE_VALUE * ONE_TERABYTE;
+		int totalInstances = 0;
+		int totalRam = 0;
+		int totalVCPU = 0;
+		int usedDisk = 0;
+		int usedInstances = 0;
+		int usedRam = 0;
+		int usedVCPU = 0;
+
+		for (Entry<String, ComputeQuota> quota : quotas.entrySet()) {
+			totalInstances += quota.getValue().getTotalQuota().getInstances();
+			totalVCPU += quota.getValue().getTotalQuota().getvCPU();
+			totalRam += quota.getValue().getTotalQuota().getRam();
+			totalQuota = new ComputeAllocation(totalVCPU, totalRam, totalInstances, totalDisk);
+
+			usedInstances += quota.getValue().getUsedQuota().getInstances();
+			usedVCPU += quota.getValue().getUsedQuota().getvCPU();
+			usedRam += quota.getValue().getUsedQuota().getRam();
+			usedDisk += quota.getValue().getUsedQuota().getDisk();
+			usedQuota = new ComputeAllocation(usedVCPU, usedRam, usedInstances, usedDisk);
+
+			result = new ComputeQuota(totalQuota, usedQuota);
```

Comment:
what do u mean with this? For each loop you chance the result to a new object and return the last one later. Thats it?

Reply:
Really not very clear, I'll fix this.